src/server/services/wms/qgsmaprendererjobproxy.cpp

```diff
@@ -56,6 +56,13 @@ namespace QgsWms
       renderJob.setFeatureFilterProvider( mFeatureFilterProvider );
 #endif
       renderJob.start();
+
+      // Allows the main thread to manage blocking call coming from rendering
+      // threads (see discussion in #18988).
```

Comment:
Better add a link here, we will sooner or later have a collision with pull request numbers.

Reply:
done